tests/Component/Customer/CustomerSelectorTest.php

```diff
@@ -38,7 +38,7 @@ public function testUserNotConnected()
 
         $session = $this->createMock('Symfony\Component\HttpFoundation\Session\SessionInterface');
 
-        $securityContext = $this->createMock('Symfony\Component\Security\Core\SecurityContextInterface');
+        $securityContext = $this->createMock('Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface');
```

Comment:
use ::class notation instead:

```
use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;

AuthorizationCheckerInterface::class
```

You will also need to change the construct of the class, take a look at this: https://github.com/sonata-project/SonataAdminBundle/blob/4a3dbddff7b519e83a8e54f0fe51d7dcebd1d1f4/src/Security/Handler/RoleSecurityHandler.php#L37

We removed the type hint, and added a check for the type inside. Because you cannot change type hint without being BC break

Reply:
Also you might need to add another argument, because AuthorizationChecker does not have getToken. SecurityContext was split into 2 classes, I cannot remember the other one.

You might add an optional argument, and check for the 2 signatures.

Reply:
I'll do it in a separate PR, thanks.

Reply:
@jordisala1991 Am i going in the right direction?
New constructor:
```php
    /**
     * @param CustomerManagerInterface $customerManager
     * @param SessionInterface $session
     * @param $authorizationChecker
     * @param LocaleDetectorInterface $localeDetector
     * @param TokenStorageInterface $tokenStorage
     */
    public function __construct(CustomerManagerInterface $customerManager, SessionInterface $session,
                                $authorizationChecker, LocaleDetectorInterface $localeDetector,
                                TokenStorageInterface $tokenStorage = null)
    {
        $this->customerManager = $customerManager;
        $this->session = $session;
        // NEXT_MAJOR: Move AuthorizationCheckerInterface check to method signature.
        if (!$authorizationChecker instanceof AuthorizationCheckerInterface) {
            throw new \InvalidArgumentException('Argument 1 should be an instance of Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface');
        }
        $this->authorizationChecker = $authorizationChecker;
        $this->locale = $localeDetector->getLocale();
        $this->tokenStorage = $tokenStorage;
    }
```
And config will look (adding extra argument will not crash)
```xml
        <service id="sonata.customer.selector" class="%sonata.customer.selector.class%">
            <argument type="service" id="sonata.customer.manager"/>
            <argument type="service" id="session"/>
            <argument type="service" id="security.authorization_checker"/>
            <argument type="service" id="sonata.intl.locale_detector"/>
            <argument type="service" id="security.token_storage"/>
        </service>
```

Reply:
I think you do